src/core/balance-sheet.ts

```diff
@@ -24,8 +26,13 @@ export const updatePendingLightningTransactions = async () => {
     processor: async ({ _id }, index) => {
       logger.trace("updating pending invoices for user %s in worker %d", _id, index)
       const user = await User.findOne({ _id })
-      const userWallet = await WalletFactory({ user, logger })
-      await userWallet.updatePendingInvoices()
+      const result = await Wallets.updateOnChainReceipt(user.id, logger)
```

Comment:
previously we needed the user because was a param of WalletFactory but with this new structure you can just use `_id` param and remove the extra query `const user = await User.findOne({ _id })`

Reply:
The `_id` param is an object. The `id` param is the string.

Reply:
the param of processor `processor: async ({ _id }, index) => {` not user.id what I mean is that you can remove `const user = await User.findOne({ _id })`